### ISSUE TITLE
"conda create env user/env" can't find packages

### ISSUE THREAD

**Issue by looooo** [conda/conda#3394]
I try to create an environment stored on anaconda with packages from conda-forge,... But these additional packages can't be found, although the anaconda users are stored in the .condarc file.
This is the output:
```
conda env create freecad/fc_2_build
Using Anaconda API: https://api.anaconda.org
Using Anaconda API: https://api.anaconda.org
Fetching package metadata .............
Solving package specifications: .
Error: Packages missing in current linux-64 channels: 
  - conda-forge::ca-certificates 2016.8.31 0
  - conda-forge::cairo 1.12.18 8
  - conda-forge::certifi 2016.8.31 py27_0
.....
```

**Comment by mwcraig** [comment]
**Note:** *This turns out to be a duplicate of #3369 and my case below is indeed fixed in conda 4.2/conda env 2.6 from `conda-canary`.*

To expand on this a bit, it has *nothing to do with conda-forge specifically*. I cannot create an environment with `conda env` with any channel specification in the name of the package.

For example, the environment file in [this gist](https://gist.github.com/mwcraig/41beb8c9c90303a0cd952fb226747a62) was created with `conda env export --no-builds -n ast366 > ast366.yml`, except that I manually edited the environment name to be `ast366-test`. 

When I deactivate to get back to the root environment and then try to create a new environment with this file with `conda env create -f ast366.yml`, I get this error:

```
Using Anaconda Cloud api site https://api.anaconda.org
Using Anaconda Cloud api site https://api.anaconda.org
Fetching package metadata ...............
Solving package specifications: .
Error: Packages missing in current osx-64 channels:
  - astropy::astroscrappy 1.0.3*
  - astropy::ccdproc 1.1.0*
  - astropy::reproject 0.3.1*
```

If I remove the channel part of the specification (`astropy::`) from the dependencies the environment is created fine.

**Comment by looooo** [comment]
thanks, i will wait for 4.2, and close this issue.
